core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -1164,10 +1164,9 @@ public Expression handle(Expression x) {
 
   static Expression getDefaultValue(Type type) {
     Primitive p = Primitive.of(type);
-    if (p != null) {
-      return Expressions.constant(p.defaultValue, type);
-    }
-    return Expressions.constant(null, type);
+    return p != null
+        ? Expressions.constant(p.defaultValue, type)
+        : Expressions.constant(null);
```

Comment:
thanks, why we need type info for null ? BTW all tests are green, probably you can suggest additional test for your assumption ?

Reply:
@DonnyZone , you mean e.g. if the expression is used in an overloaded method that requires a null parameter to be cast to avoid ambiguity?

Reply:
@rubenada @DonnyZone Yes, just concern, but I can't come up with a UT to trigger this code path right now. This function is also called by some complex WinAggs. I'm personally incline to perform such common optimizations at the Linq4J side when generating the **code string**, rather than dropping type information of expressions.

Reply:
On the other hand, this PR changes two static functions which can be called by users. But the functions are not only limited to the scope of this ticket. For example, the original `Expressions.makeTernary` is intended to infer the untyped null, but this PR breaks this behavior.

Reply:
@DonnyZone ok i see, i will prepare new approach soon, thanks.